WebAPI/Controllers/RolesAndPermissionsController.cs

```diff
@@ -31,11 +84,303 @@ public RolesAndPermissionsController(IRolesAndPermissionsManager rolesAndPermiss
         [ProducesResponseType(403, Type = null)]
         [ProducesResponseType(500, Type = null)]
         [HttpPost("InsertRole")]
-        public IActionResult InsertRole([FromBody] Role role)
+        public IActionResult InsertRole([FromBody] RoleDTO role)
         {
             if (role == null) return BadRequest();
-            var resRole = _rolesAndPermissionsManager.InsertRole(role);
-            return StatusCode(StatusCodes.Status201Created, resRole.Id);
+            return StatusCode(StatusCodes.Status201Created, _rolesAndPermissionsManager.InsertRole(role));
+        }
+
+        /// <summary>
+        ///     removes a role from the DB
+        /// </summary>
+        /// <param name="roleId">Role's Id to be deleted </param>
+        /// <response code="500"></response>
+        /// <response code="403">UnAuthorized Access </response>
+        /// <response code="202">Role Deleted</response>
+        [ProducesResponseType(202, Type = null)]
+        [ProducesResponseType(403, Type = null)]
+        [ProducesResponseType(500, Type = null)]
+        [HttpDelete("DeleteRole")]
+        public IActionResult DeleteRole(int roleId)
+        {
+            if (roleId <= 0) return BadRequest();
+            _rolesAndPermissionsManager.DeleteRole(roleId);
+            return StatusCode(StatusCodes.Status202Accepted);
+        }
+
+
+        /// <summary>
+        ///     Assigns Role to User
+        /// </summary>
+        /// <remarks>
+        ///     Please Make sure to spell the rolename and userId correctly
+        /// </remarks>
+        /// <param name="roleName">Role Name</param>
+        /// <param name="userId">User's Id</param>
+        /// <response code="500">RoleName or UserId is not correct </response>
+        /// <response code="403">UnAuthorized Access </response>
+        /// <response code="200"></response>
+        [ProducesResponseType(200, Type = null)]
+        [ProducesResponseType(403, Type = null)]
+        [ProducesResponseType(500, Type = null)]
+        [HttpGet("AssignRoleToUser")]
+        public IActionResult AssignRoleToUser(string roleName, int userId)
+        {
+            _rolesAndPermissionsManager.AssignRoleToUser(roleName, userId);
+            return StatusCode(StatusCodes.Status200OK);
+        }
+
+        /// <summary>
+        ///     removes Role from User
```

Comment:
Inconsistent letter capitalization makes the API look less professional.
This is a recurring issue in your code. Please review your controller